File: ThesisBackend/Models/Race.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ThesisBackend.Models;

public class Race
{
	[Key]
	public int Id { get; set; }

	[Required]
	[StringLength(64)]
	public string Name { get; set; }

	public string Description { get; set; }

	public int CreatorId { get; set; }
	[ForeignKey("CreatorId")]
	public User Creator { get; set; }

	public int? CrewId { get; set; }
	[ForeignKey("CrewId")]
	public Crew Crew { get; set; }

	public RaceType RaceType { get; set; }

	[StringLength(128)]
	public string Location { get; set; }

	[StringLength(32)]
	public string Coordinates { get; set; }

	public bool Private { get; set; }

	public List<User> Users { get; set; } = new List<User>(); // Many-to-many with User
}

File: rev-n-roll/src/app/models/race.ts
export enum RaceType {
  Drag = 0,
  Circuit = 1,
  Sprint = 2
}

export interface RaceUser {
  id: number;
  nickname: string;
}

export interface Race {
  id: number;
  name: string;
  description: string;
  creatorId: number;
  crewId?: number;
  raceType: RaceType;
  location: string;
  coordinates: string;
  date: string;
  private: boolean;
  users: RaceUser[];
}

export interface RaceRequest {
  name: string;
  description: string;
  crewId?: number;
  raceType: RaceType;
  location: string;
  coordinates: string;
  date: string;
  private: boolean;
}